app/controllers/api/messaging.php

```diff
@@ -0,0 +1,2138 @@
+    ->param('enabled', true, new Boolean(), 'Set as enabled.', true)
+    ->param('username', '', new Text(0), 'Telesign username.')
+    ->param('password', '', new Text(0), 'Telesign password.')
+    ->inject('dbForProject')
+    ->inject('response')
+    ->action(function (string $providerId, string $name, bool $default, bool $enabled, string $username, string $password, Database $dbForProject, Response $response) {
+        $providerId = $providerId == 'unique()' ? ID::unique() : $providerId;
+        $provider = new Document([
+            '$id' => $providerId,
+            'name' => $name,
+            'provider' => 'telesign',
+            'type' => 'sms',
+            'search' => $providerId . ' ' . $name . ' ' . 'telesign' . ' ' . 'sms',
+            'default' => $default,
+            'enabled' => $enabled,
+            'credentials' => [
+                'username' => $username,
+                'password' => $password,
+            ],
+        ]);
+
+        // Check if a default provider exists, if not, set this one as default
+        if (
+            empty($dbForProject->findOne('providers', [
+                Query::equal('default', [true]),
+                Query::equal('type', ['sms'])
+            ]))
+        ) {
+            $provider->setAttribute('default', true);
+        }
+
+        try {
+            $provider = $dbForProject->createDocument('providers', $provider);
+        } catch (DuplicateException) {
+            throw new Exception(Exception::PROVIDER_ALREADY_EXISTS);
+        }
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic($provider, Response::MODEL_PROVIDER);
+    });
+
+App::post('/v1/messaging/providers/textmagic')
+    ->desc('Create Textmagic Provider')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'providers.create')
+    ->label('audits.resource', 'providers/{response.$id}')
+    ->label('scope', 'providers.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'createTextmagicProvider')
+    ->label('sdk.description', '/docs/references/messaging/create-textmagic-provider.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_PROVIDER)
+    ->param('providerId', '', new CustomId(), 'Provider ID. Choose a custom ID or generate a random ID with `ID.unique()`. Valid chars are a-z, A-Z, 0-9, period, hyphen, and underscore. Can\'t start with a special char. Max length is 36 chars.')
+    ->param('name', '', new Text(128), 'Provider name.')
+    ->param('default', false, new Boolean(), 'Set as default provider.', true)
+    ->param('enabled', true, new Boolean(), 'Set as enabled.', true)
+    ->param('username', '', new Text(0), 'Textmagic username.')
```

Comment:
```suggestion
    ->param('username', '', new Text(0), 'TextMagic username.')
```

Reply:
done this

Reply:
@fanatic75 wait...so sorry...did they rebrand or something?? Their website now says Textmagic 😬 Their [docs](https://docs.textmagic.com/) say TextMagic, though!!

Reply:
Looks like a rebrand, I would trust their main site to be most recent vs docs. Let's stick with `Textmagic`